e2e/machine_migration_test.go

```diff
@@ -237,6 +256,179 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
 			})
 		})
 	})
+
+	var _ = Describe("Update MAPI Machine", Ordered, func() {
+		Context("with spec.authoritativeAPI modification", func() {
+			It("should allow modification of authoritativeAPI from ClusterAPI to MachineAPI", func() {
+				By("Attempting to modify authoritativeAPI from ClusterAPI to MachineAPI")
+
+				// Create a new machine for this test to avoid affecting other tests
+				testMachineName := "machine-auth-change-83955"
+				testMachine := createMAPIMachineWithAuthority(ctx, cl, testMachineName, machinev1beta1.MachineAuthorityClusterAPI)
+
+				_, err := capiframework.GetMachine(cl, testMachineName, capiframework.CAPINamespace)
+				Expect(err).ToNot(HaveOccurred(), "CAPI Machine should be created")
+				// Ensure any pending updates are processed
+				Eventually(func() error {
+					// Get the latest version again right before update
+					currentMachine, err := mapiframework.GetMachine(cl, testMachineName)
+					if err != nil {
+						return err
+					}
+
+					updatedMachine := currentMachine.DeepCopy()
+					updatedMachine.Spec.AuthoritativeAPI = machinev1beta1.MachineAuthorityMachineAPI
+
+					err = cl.Update(ctx, updatedMachine)
+					return err
+				}, capiframework.WaitMedium, capiframework.RetryMedium).Should(Succeed(), "Should allow modification of authoritativeAPI")
+
+				// Verify the machine still exists after the change (don't wait for specific status)
+				_, err = mapiframework.GetMachine(cl, testMachineName)
+				Expect(err).ToNot(HaveOccurred(), "Machine should still exist after authoritativeAPI change")
+
+				// Clean up the test machine
+				DeferCleanup(func() {
+					By("Cleaning up test machine")
+					// Try to delete the MAPI machine first
+					if testMachine != nil {
+						mapiframework.DeleteMachines(ctx, cl, testMachine)
+					}
```

Comment:
Is there any reason we're not using `cleanupMachineResources(` as we do below?

Reply:
The test was failing , for cleaning up , some conflict due to machine already in deleted status .. , to check that , just deleted mapi and it was cleaning up properly, we shall use cleanupMachineResources , let me check if that is some bug , or was just test problem.

Reply:
In the above context only using one `testMachine`, hence cannot use the cleanresources as is , which is used below.

Reply:
I'm not sure I'm following here 😓 can we see some logs? My reading of `cleanupMachineResources` shows that we're basically running the same code here and in the function..

Reply:
I meant the cleanupresources taking args as capi and mapi machines) - , where as in here , we just check that we can modify authoritativeAPI field and then immediately clean up the mapi machine that is updated , so not needed to explicitly try to delete capimachinename as well , which is why simply deleting the updated mapi machine only.
`						cleanupMachineResources(
							ctx,
							cl,
							[]*clusterv1.Machine{testCapiMachine},
							[]*machinev1beta1.Machine{testMapiMachine},
						)`